src/social-network/PostInfo.ts

```diff
@@ -41,7 +43,7 @@ export abstract class PostInfo {
      * The un-decrypted post content.
      * It MUST be the original result (but can be updated by the original parser).
      */
-    readonly parsedPostContent = new ValueRef<TypedMessage>(makeTypedMessageCompound([]), isTypedMessageEqual)
+    readonly parsedPostContent = new ValueRef<TypedMessageCompound>(makeTypedMessageCompound([]), isTypedMessageEqual)
```

Comment:
Why change from `TypedMessage` to a concreate type?

Reply:
It's easier to be processed by plugins in this way.

Reply:
That's not a good reason to break the abstraction.

Reply:
A post can be understood as sequential messages.